karma.conf.js

```diff
@@ -0,0 +1,56 @@
+var webpackConfig = require("./webpack.config")[0];
+const path = require("path");
+Object.assign(webpackConfig, {
+    devtool: "inline-source-map",
+    externals: [
+        "react/lib/ExecutionEnvironment",
+        "react/lib/ReactContext",
+        "react/addons",
+        "jsdom"
+    ]
+});
+
+module.exports = function (config) {
+    if (config.codeCoverage) {
+        Object.assign(webpackConfig, {
+            module: Object.assign(webpackConfig.module, {
+                rules: webpackConfig.module.rules.concat([{
+                    test: /\.ts$/,
+                    include: path.resolve(__dirname, "src"),
+                    loader: "istanbul-instrumenter-loader",
+                    enforce: "post",
+                    exclude: /\.(spec)\.ts$/
+                }])
+            })
+        });
+    }
+
+    config.set({
+        basePath: "",
+        frameworks: ["jasmine"],
+        files: [
+            { pattern: "src/**/*.ts", included: false, served: false, watched: true },
+            { pattern: "tests/**/*.ts", included: false, served: false, watched: true },
+            "tests/test-index.js"
+        ],
+        exclude: [],
+        preprocessors: { "tests/test-index.js": ["webpack", "sourcemap"] },
```

Comment:
Add space before `[` and space before `]`

Reply:
Done